runtime/cudaq/qis/managers/photonics/PhotonicsExecutionManager.cpp

```diff
@@ -19,6 +19,68 @@
 
 namespace cudaq {
 
+struct PhotonicsState : public cudaq::SimulationState {
+  /// @brief The state. This class takes ownership move semantics.
+  qpp::ket state;
+
+  PhotonicsState(qpp::ket &&data) : state(std::move(data)) {}
+
+  std::size_t getNumQubits() const override {
+    throw "not supported for this photonics simulator";
+  }
+
+  std::complex<double> overlap(const cudaq::SimulationState &other) override {
+    throw "not supported for this photonics simulator";
+  }
+
+  std::complex<double>
+  getAmplitude(const std::vector<int> &basisState) override {
+    /// TODO: Check basisState.size() matches qudit count
+
+    // Convert the basis state to an index value
+    const std::size_t idx = std::accumulate(
+        std::make_reverse_iterator(basisState.end()),
+        std::make_reverse_iterator(basisState.begin()), 0ull,
+        [](std::size_t acc, int bit) { return (acc << 1) + bit; });
+    return state[idx];
```

Comment:
I thought this logic was for binary states (`basisState` is a vector of 0 or 1). For qudit states of dimension `d`, it should be `acc * d + bit`.

Reply:
Thanks, Thien! Addressed this in f6113fd40becb08d47c273518794ae1524da640c.